From 44f1f0303127f75a382c5373fe40094d42d144b7 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Fri, 5 Jun 2020 20:05:31 +0200
Subject: [PATCH 21/35] Support unexpected onAuthenticated

Xiaomi Mi 8 SE send onAuthenticated during enroll
(NO I HAVE NO IDEA WHY)
In that case performance stats object doesn't exist (yet), ignore.
---
 .../server/biometrics/BiometricServiceBase.java        | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/BiometricServiceBase.java b/services/core/java/com/android/server/biometrics/BiometricServiceBase.java
index ec26f9d0cf8..3fd188725bc 100644
--- a/services/core/java/com/android/server/biometrics/BiometricServiceBase.java
+++ b/services/core/java/com/android/server/biometrics/BiometricServiceBase.java
@@ -732,10 +732,12 @@ public abstract class BiometricServiceBase extends SystemService
         if (client != null && client.onAuthenticated(identifier, authenticated, token)) {
             removeClient(client);
         }
-        if (authenticated) {
-            mPerformanceStats.accept++;
-        } else {
-            mPerformanceStats.reject++;
+        if(mPerformanceStats != null) {
+            if (authenticated) {
+                mPerformanceStats.accept++;
+            } else {
+                mPerformanceStats.reject++;
+            }
         }
     }
 
-- 
2.25.1

